.github/scripts/check-cross-version-deps.sc

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env -S scala-cli shebang
```

Comment:
There is no actual mill taks to do it? :O

Reply:
Didn't find anything... there's `ivyDepsTree --whatDependsOn` to make a specific check, but to do a general one, I didn't find a better way.

And it seems people looked for one before me:
- https://github.com/VirtusLab/scala-cli/issues/2421#issuecomment-1733787851
- https://github.com/VirtusLab/scala-cli/pull/2541#issuecomment-1812690252

Reply:
I guess another way would be to set up a dedicated Mill plugin 😅

Reply:
An alternative is to have a small sbt test project that has the dependency on cli and we run it for latest nightly

Reply:
I'd rather be able to run the check on a PR, though.
The script is simple enough, too, I think.

Reply:
It has regexes :P that is never simple

Reply:
🤷‍♂️